### PR TITLE
TASK-4495 Cohort overview detailed tab samples table needs to be paginated

### PR DESCRIPTION
**Pull request by Rodielm**
This PR involves the implementation of a new datatable that includes additional features such as pagination, sorting, and filtering.

### CODE REVIEW

**Comment by Rodielm** [comment]
Task linked: [TASK-4495 Cohort overview detailed tab samples table needs to be paginated](https://app.clickup.com/t/36631768/TASK-4495)

File: src/webcomponents/cohort/cohort-view.js [hunk 1]
```diff
@@ -113,6 +122,51 @@ export default class CohortView extends LitElement {
...
+            search: true,
+            searchAlign: "left",
+            pagination: true,
+            pageSize: 10,
+            pageList: [10, 25, 50],
+            gridContext: this,
+            formatLoadingMessage: () => "<div><loading-spinner></loading-spinner></div>",
+        });
+    }
+
```
Comment (jmjuanes) [inline src/webcomponents/cohort/cohort-view.js:169]: Please remove this.

File: src/webcomponents/cohort/cohort-view.js [hunk 2]
```diff
@@ -225,42 +279,38 @@ export default class CohortView extends LitElement {
...
+                                        id: "sample",
                                         title: "Samples ID",
                                         field: "id",
+                                        sortable: true,
                                     },
                                     {
+                                        id: "somatic",
                                         title: "Somatic",
                                         field: "somatic",
-                                        defaultValue: "false",
```
Comment (jmjuanes) [inline src/webcomponents/cohort/cohort-view.js:249]: Please add a formatter for displaying `"true"` or `"false"` as strings.

File: src/webcomponents/individual/individual-view.js [hunk 3]
```diff
@@ -316,16 +316,12 @@ export default class IndividualView extends LitElement {
                                     {
                                         title: "Somatic",
                                         field: "somatic",
-                                        defaultValue: "false",
+                                        fomatter: value => value ?? false
```
Comment (jmjuanes) [inline src/webcomponents/individual/individual-view.js:319]: Please change the returned value to string: 

```
formatter: value => value ? "true" : "false",
```

File: src/webcomponents/individual/individual-view.js [hunk 4]
```diff
@@ -316,16 +316,12 @@ export default class IndividualView extends LitElement {
...
                                     },
                                     {
                                         title: "Phenotypes",
                                         field: "phenotypes",
-                                        type: "custom",
-                                        defaultValue: "-",
-                                        display: {
-                                            render: data => data?.length ? html`${data.map(d => d.id).join(", ")}` : "-",
-                                        },
+                                        formatter: value => value?.length ? html`${value.map(d => d.id).join(", ")}` : "-",
```
Comment (jmjuanes) [inline src/webcomponents/individual/individual-view.js:324]: Please remove the `html` from the formatter function

File: src/webcomponents/user/user-projects.js [hunk 5]
```diff
@@ -167,32 +167,25 @@ export default class UserProjects extends LitElement {
...
-                                type: "custom",
                                 field: "id",
-                                display: {
-                                    render: id => html`
-                                        <a href="#browser/${project.id}/${id}" title="Variant Browser" style="white-space:nowrap;">
+                                formatter: (value, row) => `
+                                        <a href="#browser/${row.id}/${value}" title="Variant Browser" style="white-space:nowrap;">
                                             <i class="fas fa-external-link-alt icon-padding"></i> VB
                                         </a>
                                     `,
```
Comment (jmjuanes) [inline src/webcomponents/user/user-projects.js:188]: Please fix indentation

File: src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js [hunk 6]
```diff
@@ -206,48 +206,50 @@ class VariantInterpreterQcSummary extends LitElement {
...
-                                            `,
-                                        }
+                                        field: "file",
+                                        formatter: (value, row) => html`
+                                            <div>
+                                                <span style="font-weight: bold">
+                                                    ${value}
+                                                </span>
+                                            </div>
+                                        `,
```
Comment (jmjuanes) [inline src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js:216]: Please remove the `html` from this formatter

File: src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js [hunk 7]
```diff
@@ -206,48 +206,50 @@ class VariantInterpreterQcSummary extends LitElement {
...
-                                            render: data => html`
-                                                <div>${data.sdInsertSize}</div>
-                                            `,
-                                        }
+                                        field: "sdInsertSize",
+                                        formatter: value => html`
+                                            <div>
+                                                ${value}
+                                            </div>
+                                        `,
```
Comment (jmjuanes) [inline src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js:225]: Please remove the `html` from this formatter

File: src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js [hunk 8]
```diff
@@ -206,48 +206,50 @@ class VariantInterpreterQcSummary extends LitElement {
...
-                                            render: data => html`
-                                                <div>${data.avgInsertSize}</div>
-                                            `,
-                                        }
+                                        field: "avgInsertSize",
+                                        formatter: value => html`
+                                            <div>
+                                                ${value}
+                                            </div>
+                                        `,
```
Comment (jmjuanes) [inline src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js:234]: Please remove the `html` from this formatter.

File: src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js [hunk 9]
```diff
@@ -206,48 +206,50 @@ class VariantInterpreterQcSummary extends LitElement {
...
-                                        type: "custom",
-                                        display: {
-                                            render: data => html`
-                                                <div>${data.duplicateReadRate}</div>
+                                        field: "duplicateReadRate",
+                                        formatter: value => html`
+                                                <div>
+                                                    ${value}
+                                                </div>
                                             `,
```
Comment (jmjuanes) [inline src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js:243]: Please remove the `html` from this formatter and fix indentation.

File: src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js [hunk 10]
```diff
@@ -206,48 +206,50 @@ class VariantInterpreterQcSummary extends LitElement {
...
-                                            render: data => html`
-                                                <div>${data.avgSequenceDepth}</div>
-                                            `,
-                                        }
+                                        field: "avgSequenceDepth",
+                                        formatter: value => html`
+                                            <div>
+                                                ${value}
+                                            </div>
+                                        `,
```
Comment (jmjuanes) [inline src/webcomponents/variant/interpretation/variant-interpreter-qc-summary.js:252]: Please remove the `html` from this formatter.

**Review by jmjuanes — CHANGES_REQUESTED** [review]

**Review by jmjuanes — APPROVED** [review]

**Pull request merged**